api/src/files/files.controller.ts

```diff
@@ -0,0 +1,120 @@
+import { UpdateStudyDocDto } from './dto/update-study-doc.dto';
+import type { StudyDoc } from './schemas/study-doc.schema';
+import { FilesService } from './services/file-uploads.service';
+import { StudyDocsService } from './services/study-docs.service';
+
+@UseGuards(JwtAuthGuard)
+@Controller({ path: 'files', version: '1' })
+export class FilesController {
+  constructor(private readonly studyDocsService: StudyDocsService, private readonly filesService: FilesService) {}
+
+  // Create
+  @UseInterceptors(FileInterceptor('file', { limits: { fileSize: uploadConfig.maxSize } }))
+  @Post('/study-docs/upload')
+  public async uploadCourseDocument(@CurrentUser() user: User,
+  @Body() body: Partial<CreateFileUploadDto & CreateStudyDocDto>,
+  @UploadedFile() file: Express.Multer.File): Promise<StudyDoc> {
+    if (!file)
+      throw new BadRequestException('No file has been provided.');
+
+    const fileDocument = await this.filesService.create(user, {
+      originalName: file.originalname,
+      fileSize: file.size,
+      fileKind: FileType.StudyDocs,
+      type: file.mimetype,
+      encoding: file.encoding,
+      lastModified: body?.lastModified ?? new Date(),
+    }, file, FileType.StudyDocs);
+
+    return await this.studyDocsService.create(user, body as CreateStudyDocDto, fileDocument);
+  }
+
+  // Essais de crud
+
+  // Read
+  @Get('/study-docs/search')
+  public async getAllUploads(
+    @Query() query: PaginateDto,
+  ): Promise<CustomPaginateResult<StudyDoc> | { items: StudyDoc[] }> {
+    if (query.page) {
+      return await this.studyDocsService.findAll({
+        page: query.page,
+        itemsPerPage: query.itemsPerPage ?? 10,
+      }) as CustomPaginateResult<StudyDoc>;
+    }
+    const items = await this.studyDocsService.findAll() as StudyDoc[];
+    return { items };
+  }
+
+  // Get file upload info
+  @Get(':id')
+  public async getOne(@Param('id', ParseIntPipe) id: number): Promise<StudyDoc> {
+    return await this.studyDocsService.findOne(id);
+  }
+
+  // Update file upload info
+  @Patch(':id')
+  public async updateDoc(
+    @CurrentUser() user: User,
+    @Param('id', ParseIntPipe) id: number,
+    @Body() updateCourseDocDto: UpdateStudyDocDto,
+  ): Promise<StudyDoc> {
+    return await this.studyDocsService.update(user, id, updateCourseDocDto);
+  }
+
+  // Delete file upload info + file (TODO)
+  @Delete(':id')
+  public async remove(@CurrentUser() user: User, @Param('id', ParseIntPipe) id: number): Promise<void> {
+    await this.studyDocsService.remove(user, id);
+  }
+
+  // Get file
+  @Get('/get-file/:id')
+  public async getFile(@Param('id', ParseIntPipe) id: string, @Response({ passthrough: true }) res: Res): Promise<StreamableFile> {
+    const fileDocument = await this.filesService.getUploadById(id);
+    if (!fileDocument)
+      throw new NotFoundException('File does not exist or has been removed from the database.');
+    else if (fileDocument.deletedDate)
```

Comment:
```suggestion
    if (fileDocument.deletedDate)
```
pas la peine d'utiliser un else ici